### DISCUSSION TITLE
TENV_AUTO_INSTALL and lock issues with cmdline tools

### DISCUSSION THREAD

**Discussion by bdeshi** [tofuutils/tenv#135] [General]
for example [startship](https://github.com/starship/starship) terraform integration calls `terraform version` to show current tf version in `$PS1`. however with TENV_AUTO_INSTALL that command starts installing matching terraform if not found, creates a lock file, meanwhile starship terminates the command because it takes more than 1s (starship timeout). so the .lock file stays there, and breaks subsequent tf installs.

i guess can there be another method to ensure auto installs without using the .lock file that can often get left behind?

**Comment by dvaumoron** [comment]
Which **tenv** version do you use ? This issue should be corrected since [v1.10.2](https://github.com/tofuutils/tenv/releases/tag/v1.10.2) (it may depend on the kind of signal you send to stop the `terraform` proxy command).

**Comment by bdeshi** [comment]
im running tenv v1.11.1

**Comment by dvaumoron** [comment]
Which kind of termination signal **starship** send to end `terraform` proxy ? A sigkill does not allow to gracefully end with `.lock` file deletion.
